lh-test/src/main/java/io/littlehorse/test/WfRunVerifier.java

```diff
@@ -114,36 +114,47 @@ public WfRunVerifier waitForUserTaskRunStatus(int threadRunNumber, int nodeRunNu
     public WfRunVerifier waitForNodeRunStatus(int threadRunNumber, int nodeRunNumber, LHStatus status) {
         Function<Object, LHStatus> objectLHStatusFunction = context -> {
             String wfRunId = context.toString();
-            return lhClientTestWrapper
-                    .getNodeRun(wfRunId, threadRunNumber, nodeRunNumber)
+            return lhClient.getNodeRun(nodeRunIdFrom(wfRunId, threadRunNumber, nodeRunNumber))
                     .getStatus();
         };
-        steps.add(new WaitForStatusStep<>(objectLHStatusFunction, status));
+        steps.add(new WaitForStatusStep<>(objectLHStatusFunction, status, steps.size() + 1));
         return this;
     }
 
     public WfRunVerifier waitForThreadRunStatus(int threadRunNumber, LHStatus threadRunStatus) {
         Function<Object, LHStatus> objectLHStatusFunction = context -> {
-            String wfRunId = context.toString();
-            ThreadRun threadRun = lhClientTestWrapper.getWfRun(wfRunId).getThreadRuns(threadRunNumber);
+            ThreadRun threadRun =
+                    lhClient.getWfRun(wfRunIdFrom(context.toString())).getThreadRuns(threadRunNumber);
             return threadRun.getStatus();
         };
-        steps.add(new WaitForStatusStep<>(objectLHStatusFunction, threadRunStatus));
+        steps.add(new WaitForStatusStep<>(objectLHStatusFunction, threadRunStatus, steps.size() + 1));
         return this;
     }
 
     public WfRunVerifier thenVerifyNodeRun(int threadRunNumber, int nodeRunNumber, Consumer<NodeRun> matcher) {
-        steps.add(new VerifyNodeRunStep(threadRunNumber, nodeRunNumber, matcher));
+        steps.add(new VerifyNodeRunStep(threadRunNumber, nodeRunNumber, matcher, steps.size() + 1));
         return this;
     }
 
     public WfRunVerifier waitForTaskStatus(int threadRunNumber, int nodeRunNumber, TaskStatus taskStatus) {
         Function<Object, TaskStatus> objectLHStatusFunction = context -> {
-            NodeRun nodeRun = lhClientTestWrapper.getNodeRun(context.toString(), threadRunNumber, nodeRunNumber);
-            TaskRun taskRun = lhClientTestWrapper.getTaskRun(nodeRun.getTask().getTaskRunId());
+            NodeRun nodeRun = lhClient.getNodeRun(nodeRunIdFrom(context.toString(), threadRunNumber, nodeRunNumber));
+            TaskRun taskRun = lhClient.getTaskRun(nodeRun.getTask().getTaskRunId());
             return taskRun.getStatus();
         };
-        steps.add(new WaitForStatusStep<>(objectLHStatusFunction, taskStatus));
+        steps.add(new WaitForStatusStep<>(objectLHStatusFunction, taskStatus, steps.size() + 1));
         return this;
     }
+
+    private WfRunId wfRunIdFrom(String wfRunId) {
+        return WfRunId.newBuilder().setId(wfRunId).build();
+    }
+
+    private NodeRunId nodeRunIdFrom(String wfRunId, int threadRunNumber, int nodeRunNumber) {
```

Comment:
I've added a ticket to the backlog to create a class `LHProtoUtil` or something like that with a bunch of methods like this. We should make life easier for customers :smile: